src/sidebar/util/oauth-client.js

```diff
@@ -245,6 +212,41 @@ class OAuthClient {
     };
     return this.$http.post(url, data, requestConfig);
   }
+
+  /**
+   * Create a pop-up window for use with `OAuthClient#authorize`.
+   *
+   * This function _must_ be called in the same turn of the event loop as the
+   * button or link which initiates login to avoid triggering the popup blocker
+   * in certain browsers. See https://github.com/hypothesis/client/issues/534
+   * and https://github.com/hypothesis/client/issues/535.
+   *
+   * @param {Window} $window - The parent of the popup window.
+   * @return {Window}
+   */
+  static createLoginPopupWindow($window) {
```

Comment:
This method not only creates the window object for, but shows (opens) the popupWindow. Worth mentioning in the comment?

Reply:
I didn't think it was possible to create a window without also showing it to the user using DOM APIs, short of using a hack like positioning it off-screen. I'll clarify this in any case.